### PR TITLE
added PCI-DSS to OCP on GCP

### PR DESCRIPTION
**Pull request by AdrianHammond**
This PR updated the OCP on GCP install docs and code to add PCI-DSS policy. I have also updated the top level readme to reflect addition of the policy.

### CODE REVIEW

File: ocp/README.md [hunk 1]
```diff
@@ -16,13 +16,13 @@ For each cloud provider documentation and working code will be provided to:
 5. Implement OCP Compliance Operator
 6. Manual Remediation of CIS Controls
 
-Using the above will provide an OCP Cluster which is compliant with most CIS Policies. There are a number of CIS policies that require a user to make local decisions on how the policy would be implemented, CIS describe these as MANUAL. 
+Using the above will provide an OCP Cluster which is compliant with most CIS and PCI-DSS Policies. There are a number of CIS and PCI-DSS policies that require a user to make local decisions on how the policy would be implemented, these are describe these as MANUAL.
```
Comment (eddie-knight) [inline ocp/README.md:19]:
```suggestion
Using the above will provide an OCP Cluster which is compliant with most CIS and PCI-DSS Policies. There are a number of CIS and PCI-DSS policies that require a user to make local decisions on how the policy would be implemented, these are described as MANUAL. 
```

File: ocp/README.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ For each cloud provider documentation and working code will be provided to:
 5. Implement OCP Compliance Operator
 6. Manual Remediation of CIS Controls
 
-Using the above will provide an OCP Cluster which is compliant with most CIS Policies. There are a number of CIS policies that require a user to make local decisions on how the policy would be implemented, CIS describe these as MANUAL. 
+Using the above will provide an OCP Cluster which is compliant with most CIS and PCI-DSS Policies. There are a number of CIS and PCI-DSS policies that require a user to make local decisions on how the policy would be implemented, these are describe these as MANUAL. 
 
 We do provide examples of how some of these types of policies could implemented, for example adding an identity provider or replacing the self signed certificates.
 
-In step 6 we have provided some guidance on remediation of CIS control. 
+In step 6 we have provided some guidance and examples on how to remediation some of the CIS controla.
```
Comment (eddie-knight) [inline ocp/README.md:23]:
```suggestion
In step 6 we have provided some guidance and examples on how to remediate some of the CIS controls. 
```

File: ocp/README.md [hunk 3]
```diff
@@ -16,13 +16,13 @@ For each cloud provider documentation and working code will be provided to:
...
-Using the above will provide an OCP Cluster which is compliant with most CIS Policies. There are a number of CIS policies that require a user to make local decisions on how the policy would be implemented, CIS describe these as MANUAL. 
+Using the above will provide an OCP Cluster which is compliant with most CIS and PCI-DSS Policies. There are a number of CIS and PCI-DSS policies that require a user to make local decisions on how the policy would be implemented, these are describe these as MANUAL. 
 
 We do provide examples of how some of these types of policies could implemented, for example adding an identity provider or replacing the self signed certificates.
 
-In step 6 we have provided some guidance on remediation of CIS control. 
+In step 6 we have provided some guidance and examples on how to remediation some of the CIS controla. 
 
-Below is a list of the CIS policies that a user will have to investigate how they should be implemented within their own organisation. 
+Below is a list of the CIS and PCI-DSS policies that a user will have to investigate and decide how they should be implemented within their own organisation.
```
Comment (eddie-knight) [inline ocp/README.md:26]: consider adding some type of title here

File: ocp/README.md [hunk 4]
```diff
@@ -53,3 +53,44 @@ ocp4-cis-secrets-consider-external-storage          MANUAL   medium
 ocp4-cis-secrets-no-environment-variables           MANUAL   medium
```
Comment (eddie-knight) [inline ocp/README.md:54]: consider adding some type of title here

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md [hunk 5]
```diff
@@ -100,10 +100,11 @@ rhcos4-moderate      57s
 rhcos4-nerc-cip      57s
 ```
 
-7. The next step is to set up the compliance scanning using the CIS profile, the default profiles that is being used will run scans on a daily base and, where possible. auto-remiate any complaince issues.
+7. The next step is to set up the compliance scanning using the CIS and PCI-DSS profile, the default profile that is being used will run scans on a daily base and, where possible. auto-remiate any complaince issues.
```
Comment (eddie-knight) [inline ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:103]:
```suggestion
7. The next step is to set up the compliance scanning using the CIS and PCI-DSS profiles. The default profile that is being used will run scans on a daily base and, where possible, auto-remediate any compliance issues.
```

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md [hunk 6]
```diff
@@ -113,21 +114,30 @@ oc get compliancesuite -w
...
 cis-compliance-ocp4   DONE          NON-COMPLIANT
 cis-compliance-ocp4   DONE          NON-COMPLIANT
+pci-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
+pci-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
+pci-compliance-ocp4   DONE          NON-COMPLIANT
+pci-compliance-ocp4   DONE          NON-COMPLIANT
 ```
 
-9. Once the scan is complete the following command can be used to check the complaince scan results, when setting up the compliance scan object (step 7), we set the complaince operator to automatical remidate compliance failures. 
+9. Once the scans are complete the following command can be used to check the complaince scan results, when setting up the compliance scan object (step 7), we set the complaince operator to automatical remidate compliance failures.
```
Comment (eddie-knight) [inline ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:140]:
```suggestion
9. Once the scans are complete, the following command can be used to check the compliance scan results. When setting up the compliance scan object (step 7), we set the compliance operator to automatically remediate compliance failures. 
```

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md [hunk 7]
```diff
@@ -137,38 +147,35 @@ An extract of the oc get ccr command can be seen below.
...
+ocp4-pci-dss-accounts-restrict-service-account-tokens                     MANUAL   medium
+ocp4-pci-dss-accounts-unique-service-account                              MANUAL   medium
+ocp4-pci-dss-api-server-admission-control-plugin-alwaysadmit              PASS     medium
+ocp4-pci-dss-api-server-admission-control-plugin-alwayspullimages         PASS     high
+ocp4-pci-dss-api-server-admission-control-plugin-namespacelifecycle       PASS     medium
+ocp4-pci-dss-api-server-admission-control-plugin-noderestriction          PASS     medium
+ocp4-pci-dss-api-server-admission-control-plugin-scc                      PASS     medium
+```
+
+Initially a number of the compliance checks will FAIL, many of them will be remediated automatically by the compliance operator. This remediation process can take sometime (hours). These remediations are actioned by the OCP cluster operators, using the following command the progress of these cluster operators can be seen.
```
Comment (eddie-knight) [inline ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:170]:
```suggestion
Initially a number of the compliance checks will fail, and many of them will be remediated automatically by the compliance operator. This process may take hours to complete. These remediations are actioned by the OCP cluster operators, using the following command the progress of these cluster operators can be seen.
```

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md [hunk 8]
```diff
@@ -137,38 +147,35 @@ An extract of the oc get ccr command can be seen below.
...
 
 '''shell
 oc get co
 ```
 Below is an example of the output of this command.
 
+#todo - update with new output when update to 4.13 takes place
```
Comment (eddie-knight) [inline ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:177]: is this TODO intended to stay in the PR? up to you, just double checking.

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md [hunk 9]
```diff
@@ -223,6 +230,23 @@ oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-cis-node
...
 ```
 
+### The OCP PCI Benchmark
+
+The OCP PCI policy consists of two policies
+
+- *ocp4-pci-dss* for Cluster polies
+- *ocp4-pci-dss-node* for Node polies
+
+As with the CIS policy the *oc get compliancecheckresults* command can be used to check for scan results, for example
```
Comment (eddie-knight) [inline ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:240]:
```suggestion
As with the CIS policy the `oc get compliancecheckresults` command can be used to check for scan results. For example
```

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md [hunk 10]
```diff
@@ -223,6 +230,23 @@ oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-cis-node
...
+
+```bash
+
+oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-pci-dss,compliance.openshift.io/check-status=MANUAL
+
+oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-pci-dss-node-worker,compliance.openshift.io/check-status=PASS
+
+```
+
 Once any remediation have be made the following commands can be used to trigger a compliance rescan. A [script](rescan.sh) to trigger a rescan of all scans has been provided.
```
Comment (eddie-knight) [inline ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:250]:
```suggestion
Once any remediation has been made, the following commands can be used to trigger a compliance rescan. A [script](rescan.sh) to trigger a rescan of all scans has been provided. 
```

**Review by eddie-knight — CHANGES_REQUESTED** [review]

**Review by AdrianHammond — COMMENTED** [review]
I have accepted suggestions - thanks @eddie-knight

File: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md (hunk 8 above)
Comment (AdrianHammond) [reply ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md:177]: Yes. thanks for highlighting.

File: ocp/README.md (hunk 3 above)
Comment (AdrianHammond) [reply ocp/README.md:26]: added

File: ocp/README.md (hunk 4 above)
Comment (AdrianHammond) [reply ocp/README.md:54]: added

**Review by eddie-knight — APPROVED** [review]

**Comment by eddie-knight** [comment]
Approved, but leaving this open since AG said he was going to take a look.

**Comment by AdrianHammond** [comment]
@abdullahgarcia  - pls could you review / merge.  Many thanks A

**Comment by abdullahgarcia** [comment]
LGTM

**Review by abdullahgarcia — APPROVED** [review]

**Pull request merged**